include/materials/material_utility.tcc

```diff
@@ -203,3 +203,29 @@ inline const Eigen::Matrix<double, 6, 1> mpm::materials::dtheta_dsigma(
   // Compute and return dtheta / dsigma
   return (dtheta_dr * ((dr_dj2 * dj2_dsigma) + (dr_dj3 * dj3_dsigma)));
 }
+
+//! Compute incremental dplastic_strain
+inline const Eigen::Matrix<double, 6, 1> mpm::materials::dplastic_strain(
+    const Eigen::Matrix<double, 6, 1>& dstress,
+    const Eigen::Matrix<double, 6, 1>& dstrain,
+    const Eigen::Matrix<double, 6, 6>& de) {
+  // Compute and return dplastic_strain
+  return (dstrain - de.inverse() * dstress);
+}
+
+//! Compute plastic deviatoric strain
+inline double mpm::materials::pdstrain(
```

Comment:
This is okay though. Each method can call this function to compute pdstrain providing they have the plastic strain.

Reply:
But this should always be correct right, since even for different methods, dstrain is always divided into dstrain_elastic and dstrain_plastic anyways. The formulation of D stiffness matrix is defined in such a way that we subtract D_elastic. So this is correct. 

It's a hassle though, for one line of code, we have to transfer a lot of variables

Reply:
@ezrayst were you replying to the previous comment?

Reply:
Yes, sorry, wrong one